src/lib.rs

```diff
@@ -520,6 +520,17 @@ fn write_profile(matches: &ArgMatches) -> Result<(), Box<dyn std::error::Error>>
 fn run_rustpython(vm: &VirtualMachine, matches: &ArgMatches) -> PyResult<()> {
     let scope = vm.new_scope_with_builtins();
     let main_module = vm.new_module("__main__", scope.globals.clone());
+    let ann_dict = vm.ctx.new_dict();
+    let main_module_dict = main_module.dict();
+
+    if main_module_dict.is_none()
+        || main_module_dict
+            .unwrap()
+            .set_item("__annotations__", ann_dict.as_object().to_owned(), vm)
+            .is_err()
+    {
+        error!("Failed to initialize __main__.__annotations__")
+    }
```

Comment:
`and_then` is available when we want to run something only when an Option or Result is Some or Ok.

```suggestion
    let main_module_dict = main_module.dict();
    main_module_dict
        .and_then(|d| {
            d.set_item(
                "__annotations__",
                vm.ctx.new_dict().as_object().to_owned(),
                vm,
            )
            .ok()
        })
        .expect("Failed to initialize __main__.__annotations__");
```

Reply:
Thanks to let me know it! awesome code 😀  🙏🏻